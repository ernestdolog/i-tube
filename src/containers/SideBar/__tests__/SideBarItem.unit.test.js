import React from 'react';
import {shallow} from 'enzyme';
import SideBarItem from '../SideBarItem';

test('renders SideBarItem', () => {
  const wrapper = shallow(
    <SideBarItem />
  );
  expect(wrapper).toMatchSnapshot();
});

test('renders SideBarItem with highlight', () => {
    const wrapper = shallow(
      <SideBarItem highlight={true}/>
    );
    expect(wrapper).toMatchSnapshot();
});

test('renders SideBarItem with icon', () => {
    const wrapper = shallow(
        <SideBarItem licon='fire'/>
    );
    expect(wrapper).toMatchSnapshot();
});

test('renders SideBarItem with label', () => {
    const wrapper = shallow(
        <SideBarItem label='Trending'/>
    );
    expect(wrapper).toMatchSnapshot();
});

test('renders SideBarItem with highlight icon label', () => {
    const wrapper = shallow(
        <SideBarItem highlight={true} licon='fire' label='Trending'/>
    );
    expect(wrapper).toMatchSnapshot();
});