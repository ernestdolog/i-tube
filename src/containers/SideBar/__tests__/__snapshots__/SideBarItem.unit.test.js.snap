// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SideBarItem 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-align-container"
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
    </span>
    <span />
  </div>
</MenuItem>
`;

exports[`renders SideBarItem with highlight 1`] = `
<MenuItem
  className="sidebar-item highlight-item"
>
  <div
    className="sidebar-item-align-container"
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
    </span>
    <span />
  </div>
</MenuItem>
`;

exports[`renders SideBarItem with highlight icon label 1`] = `
<MenuItem
  className="sidebar-item highlight-item"
>
  <div
    className="sidebar-item-align-container"
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;

exports[`renders SideBarItem with icon 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-align-container"
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
    </span>
    <span />
  </div>
</MenuItem>
`;

exports[`renders SideBarItem with label 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-align-container"
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
    </span>
    <span>
      Trending
    </span>
  </div>
</MenuItem>
`;
